### PR TITLE
[flytepropeller][flyteadmin] Compiler unknown literal type error handling

### PR DESCRIPTION
**Pull request by Future-Outlier**
## Why are the changes needed?
To avoid a crash when a new IDL type is sent to the propeller or admin.

## What changes were proposed in this pull request?
Add error handling and prevent crashes in these sections of the code.

image

## How was this patch tested?
most unit tests, and 1 Python example in the comments below.

### Setup process
flyteidl: `pip install
 git+https://github.com/flyteorg/flyte.git@2c9cbd89dce53acd1cb90cc042012637ae6c31f8#subdirectory=flyteidl`
flytekit: https://github.com/flyteorg/flytekit/pull/2741

python example:
```python
from flytekit import task, workflow, ImageSpec

flytekit_hash = "c24077bce6e63bf8df0d80dbc2c5e2ff3322bca8"

flytekit = f"git+https://github.com/flyteorg/flytekit.git@{flytekit_hash}"

image = ImageSpec(
    packages=[flytekit],
    apt_packages=["git"],
    registry="localhost:30000",
)

@task(container_image=image)
def dict_task(input: dict) -> dict:
    print("Dict fields:")
    for k, v in input.items():
        print(type(k), type(v))
        print(k, v)
    return input

@workflow
def dict_wf(input: dict) -> dict:
    return dict_task(input=input)

if __name__ == "__main__":
    from flytekit.clis.sdk_in_container import pyflyte
    from click.testing import CliRunner

    runner = CliRunner()
    path = "/Users/future-outlier/code/dev/flytekit/build/PR/JSON/demo/dict_example.py"
    result = runner.invoke(pyflyte.main,
                    ["run",
                            path,
                            "dict_wf",
                            "--input",
                            '{"a": 1, "b": 2, "c": [1, 2, 3], "d": {"e": 1, "f": [1, 2, 3]}}'])
    print("Local Execution: ", result.output)
    #
    result = runner.invoke(pyflyte.main,
                           ["run",
                            "--remote",
                            path,
                            "dict_wf",
                            "--input",
                            '{"a": 1, "b": 2, "c": [1, 2, 3], "d": {"e": 1, "f": [1, 2, 3]}}'])
    print("Remote Execution: ", result.output)
```

### Screenshots
image

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Related PRs
https://github.com/flyteorg/flyte/pull/5612

### CODE REVIEW

**Comment by Future-Outlier** [comment]
todo: 
for propeller, do this
```go
if inputType == nil {
    errs.Collect(errors.NewMismatchingTypesErr(nodeID, inputVar, v.Type.String(), "Unknown Type"))
    continue
}
```

File: flyteadmin/pkg/manager/impl/validation/execution_validator.go [hunk 1]
```diff
@@ -98,6 +99,17 @@ func CheckAndFetchInputsForExecution(
...
+				return nil, errors.NewFlyteAdminErrorf(codes.InvalidArgument,
+					fmt.Sprintf(
+						"invalid %s input wrong type.\n"+
+							"Expected %s, but got %s.\n"+
+							"Suggested solution: Please update all of your Flyte images to the latest version and try"+
+							" again.",
+						name, expectedInput.GetVar().GetType().String(), inputType,
+					),
+				)
+			}
```
Comment (Future-Outlier) [inline flyteadmin/pkg/manager/impl/validation/execution_validator.go:112]: Example
```python
from dataclasses import dataclass, fields
from typing import Dict
from flytekit.types.file import FlyteFile

from flytekit import task, workflow, ImageSpec

flytekit_hash = "a964d7d5461fd79f3f8442c433825bcba830f32d"
flyteidl_hash = "0b32845e92cfe354c197ddf5ae3c8605a17f079b"

json_idl = f"git+https://github.com/flyteorg/flyte.git@{flyteidl_hash}#subdirectory=flyteidl"
flytekit = f"git+https://github.com/flyteorg/flytekit.git@{flytekit_hash}"

image = ImageSpec(
    packages=[json_idl, flytekit],
    apt_packages=["git"],
    registry="futureoutlier",
)

@dataclass
class DC_inside:
    a: int
    b: float
    # ff: FlyteFile

@dataclass
class DC:
    a: int
    b: float
    c: str
    d: Dict[str, int]
    e: DC_inside


@task(container_image=image)
def dataclass_task(input: DC) -> DC:
    def print_dataclass_info(instance, indent=0):
        for field in fields(instance):
            value = getattr(instance, field.name)
            print(f"{' ' * indent}Field: {field.name}, Type: {field.type}, Value: {value}")
            if isinstance(value, dict):
                for k, v in value.items():
                    print(f"{' ' * (indent + 4)} Type: {type(k)}, Key: {k}")
                    print(f"{' ' * (indent + 4)} Type: {type(v)}, Value: {v}")

    print("DC fields:")
    print_dataclass_info(input)

    return input

@task(container_image=image)
def dataclass_attr_dict_task(a: float, d: Dict[str, int]):
    print("DC Attr:")
    print("a:", type(a), a)
    for k, v in d.items():
        print(type(k), k)
        print(type(v), v)

@task(container_image=image)
def dataclass_attr_dc_task(dc: DC_inside):
    print("DC Attr:")
    print("a:", type(dc.a), dc.a)
    print("b:", type(dc.b), dc.b)

@task(container_image=image)
def dataclass_attr_dc_int_task(a: int):
    print("DC Attr:")
    print("a:", type(a), a)

@task(container_image=image)
def dataclass_attr_dc_float_task(b: float):
    print("DC Attr:")
    print("b:", type(b), b)

@task(container_image=image)
def dataclass_attr_dc_flytefile_task(ff: FlyteFile):
    print("FlyteFile DC Attr:")
    with open(ff, newline="") as nf:
        print(nf.read())

@workflow
def dataclass_wf(input: DC) -> DC:
    dataclass_attr_dict_task(a=input.b, d=input.d,)
    dataclass_attr_dc_task(dc=input.e)
    dataclass_attr_dc_int_task(a=input.e.a)
    dataclass_attr_dc_float_task(b=input.e.b)
    # dataclass_attr_dc_flytefile_task(ff=input.e.ff)
    return dataclass_task(input=input)


if __name__ == "__main__":
    from flytekit.clis.sdk_in_container import pyflyte
    from click.testing import CliRunner

    runner = CliRunner()
    path = "/Users/future-outlier/code/dev/flytekit/build/PR/JSON/demo/dataclass_example.py"
    result = runner.invoke(pyflyte.main, ["run", path, "dataclass_wf", "--input",
                                          '{"a": 1, "b": 3.14, "c": "example_string", "d": {"1": 100, "2": 200}, "e": {"a": 1, "b": 3.14}}'])

    print("Local Execution: ", result.output)

    result = runner.invoke(pyflyte.main, ["run", "--remote", path, "dataclass_wf", "--input",
                                          '{"a": 1, "b": 3.14, "c": "example_string", "d": {"1": 100, "2": 200}, "e": {"a": 1, "b": 3.14}}'])
    print("Remote Execution: ", result.output)
```

image

File: flyteadmin/pkg/manager/impl/validation/signal_validator.go [hunk 2]
```diff
@@ -76,6 +77,22 @@ func ValidateSignalSetRequest(ctx context.Context, db repositoryInterfaces.Repos
 	if err != nil {
 		return err
 	}
+	err = propellervalidators.ValidateLiteralType(valueType)
+	if err != nil {
+		return errors.NewFlyteAdminErrorf(codes.InvalidArgument,
```
Comment (pingsutw) [inline flyteadmin/pkg/manager/impl/validation/signal_validator.go:82]: Could we add a new error like [this](https://github.com/flyteorg/flyte/blob/bed761c33c40af23750467c828afea553c0b80a0/flyteadmin/pkg/errors/errors.go#L130) and return the new one here?

Comment (Future-Outlier) [reply flyteadmin/pkg/manager/impl/validation/signal_validator.go]: done, thank you

**Review by pingsutw — APPROVED** [review]

**Comment by Future-Outlier** [comment]
Hi, @eapolinario 
Do you want to take a look?

**Pull request merged**